added aggregation, discovery and upload end points

goegges1 opened the pull request:
Closes #1

goegges committed:
added plyr to the dependencies

goegges committed:
deleted unused manual file

goegges committed:
deleted requestModel and seperated its content to discoveredModel and aggregationData. added comments and references to the Lana API.

Commit:
renamed action column for activity statistics

goegges committed:
added error handling with stop and error message (Text + status code)

goegges committed:
initial push, needs to be checked (LANA_R-API_Upload_Example.R didn't work with a 404)

goegges committed:
initial push of download and upload.

goegges committed:
Download for csv in the working directory

goegges committed:
csvDownload added comment,
uploadCsvToLana changed to work against new API and changed name

goegges committed:
added error messages

thobai commented on R/aggregationData.R:
```diff
@@ -0,0 +1,48 @@
+#' @name getAggregationRequestData
+
+getAggregationRequestData <- function(rqBody){
+checkAuthentication()
```
indention is missing

thobai commented on R/aggregationData.R:
```diff
@@ -0,0 +1,48 @@
+if(userInfo["status_code"] == 400) {
+  stop(httr::paste0("Invalid rqBody JSON " + "Statuscode: ", userInfo["status_code"]))
+}
+
```
unnecessary blank lines / code style

thobai commented on R/aggregationData.R:
```diff
@@ -0,0 +1,48 @@
+}
+
+# Make request LANA
+aggregationRequestData <- httr::GET(paste0(lanaApiUrl, "/api/aggregatedData?request=", URLencode(rqBody)),
```
`reserved = T` is missing for the encoding

thobai commented on R/csvDownload.R:
```diff
@@ -0,0 +1,16 @@
+#' @title \code Download example CSV file
+#' @description download the incident example CSV file from our homepage to experiment with Lana
```
shouldn't use `our` here but refer to Lana Labs

thobai commented on R/csvDownload.R:
```diff
@@ -0,0 +1,16 @@
+#' @title \code Download example CSV file
```
there is no `code` in this title. Should be removed

thobai commented on R/aggregationData.R:
```diff
@@ -0,0 +1,48 @@
+#' @return total of the aggregated data
+#' @param discoveredModelData
+#' @name getAggregationRequestData
+
```
no new line between comments and function declaration

thobai commented on R/discoveredModel.R:
```diff
@@ -0,0 +1,62 @@
+#' @return discovered model
+#' @param rqBody
+#' @name getDiscoveredModel
+#' @examples
```
if examples is added it shouldn't be empty

thobai commented on R/discoveredModel.R:
```diff
@@ -0,0 +1,62 @@
+userInfo <- httr::GET(paste0(lanaApiUrl, "/api/userInfo"), httr::add_headers(Authorization = lanaAuthorization))
+
+if(userInfo["status_code"] == 400) {
+stop(httr::paste0("Invalid rqBody JSON " + "Statuscode: ", userInfo["status_code"]))
```
indention and code style needs to be changed

thobai commented on R/discoveredModel.R:
```diff
@@ -0,0 +1,62 @@
+#' @title \code Get activity performance statistics
+#' @description a method that gets the activity performance statistics
+#' @return activity performance statistics
+#' @param discoveredModelData
```
link to function to get the discoveredModelData

thobai commented on R/uploadCsvToLana.R:
```diff
@@ -0,0 +1,42 @@
+
+  if(responseLogCreation["status_code"] == 400) {
+    stop(httr::paste0("Invalid rqBody JSON " + "Statuscode: ", responseLogCreation["status_code"]))
+  }
```
code style

thobai reviewed (changes requested):
Code style needs to be improved and minor issues resolved

thobai commented on R/discoveredModel.R:
```diff
@@ -0,0 +1,62 @@
+lanaAuthorization <- Sys.getenv("LANA_TOKEN")
+lanaUserId <- Sys.getenv("LANA_USERID")
+
+userInfo <- httr::GET(paste0(lanaApiUrl, "/api/userInfo"), httr::add_headers(Authorization = lanaAuthorization))
```
Unnecessary call to get `userInfo`. This is done in `checkAuthentication`. Checks should be made for the `getDiscoveredModelWithFilter` endpoint instead

thobai replied:
HTTP error checking should be moved into a function to be reused

Commit:
code style improvements;
moved error checking into function;
smaller improvements on documentation;
fix for URLencode options (include `reserved = T`)

Commit:
added DS_Store to gitignore

Commit:
changed error handling to use httr error handling and include messages from backend

Commit:
fixed httr package scope

Commit:
improved error checking and data aggregation

Commit:
added functions to build aggregation settings JSON and mining request JSON

Commit:
minor clean up

Commit:
improved comments

Commit:
updated README

Commit:
updated README

Commit:
Cleanup of function documentation

thobai reviewed (approved):
All necessary changes applied
